lightning/src/ln/peers/handshake/mod.rs

```diff
@@ -392,3 +392,374 @@ impl PeerHandshake {
 		Sha256::from_engine(sha).into_inner()
 	}
 }
+
+#[cfg(test)]
+mod test {
+	use hex;
+
+	use bitcoin::secp256k1;
+	use bitcoin::secp256k1::key::{PublicKey, SecretKey};
+
+	use ln::peers::handshake::PeerHandshake;
+	use ln::peers::handshake::acts::Act;
+	use ln::peers::handshake::states::HandshakeState;
+
+	struct TestCtx {
+		outbound_handshake: PeerHandshake,
+		outbound_public_key: PublicKey,
+		inbound_handshake: PeerHandshake,
+		inbound_public_key: PublicKey
+	}
+
+	impl TestCtx {
+		fn new() -> TestCtx {
+			let curve = secp256k1::Secp256k1::new();
+
+			let outbound_private_key = SecretKey::from_slice(&[0x_11_u8; 32]).unwrap();
+			let outbound_public_key = PublicKey::from_secret_key(&curve, &outbound_private_key);
+			let outbound_ephemeral_key = SecretKey::from_slice(&[0x_12_u8; 32]).unwrap();
+
+			let inbound_private_key = SecretKey::from_slice(&[0x_21_u8; 32]).unwrap();
+			let inbound_public_key = PublicKey::from_secret_key(&curve, &inbound_private_key);
+			let inbound_ephemeral_key = SecretKey::from_slice(&[0x_22_u8; 32]).unwrap();
+
+			let outbound_handshake = PeerHandshake::new_outbound(&outbound_private_key, &inbound_public_key, &outbound_ephemeral_key);
+			let inbound_handshake = PeerHandshake::new_inbound(&inbound_private_key, &inbound_ephemeral_key);
+
+			TestCtx {
+				outbound_handshake,
+				outbound_public_key,
+				inbound_handshake,
+				inbound_public_key
+			}
+		}
+	}
+
+	macro_rules! assert_matches {
+		($e:expr, $state_match:pat) => {
+			match $e {
+				$state_match => (),
+				_ => panic!()
+			}
+		}
+	}
+
+	macro_rules! do_process_act_or_panic {
+		($handshake:expr, $input:expr) => {
+			$handshake.process_act($input).unwrap().0.unwrap().serialize()
+		}
+	}
+
+	// Default Outbound::Uninitiated
+	#[test]
+	fn peer_handshake_new_outbound() {
+		let test_ctx = TestCtx::new();
+
+		assert_matches!(test_ctx.outbound_handshake.state, Some(HandshakeState::Uninitiated));
+		assert_eq!(test_ctx.outbound_handshake.get_remote_pubkey(), Some(test_ctx.inbound_public_key));
+	}
+
+	// Default Inbound::AwaitingActOne
+	#[test]
+	fn peer_handshake_new_inbound() {
+		let test_ctx = TestCtx::new();
+
+		assert_matches!(test_ctx.inbound_handshake.state, Some(HandshakeState::AwaitingActOne(_)));
+		assert!(test_ctx.inbound_handshake.get_remote_pubkey().is_none());
+	}
```

Comment:
Your reasoning seems fine. To help the next contributor, I would suggest creating a test style guide and then modifying/enforcing that style for existing code or new contributions. As a new contributor, I looked for uniformity in the existing codebase. Since it didn't exist I found something that made sense to me. I would expect others to do the same without more guidelines.

You will notice in later patches that the Transport unit tests utilize comments for the given/when/then pattern to help readers understand the tests. Would likely want to be cleaned up to match your expectations.

I agree that test naming consistency should exist for this project. I just wonder whether or not one-off changes like you suggest here will raise the quality bar long term over a more specific style guide that you can point to for new developers.

Either way, easy to sed all of this to match what you want.